src/bubo-ht.h

```diff
@@ -111,53 +79,33 @@ class BuboHashSet {
         assert(entry_buf);
         uint32_t idx = hash(entry_buf, entry_len) % table_size_;
 
-        return find_at(&table_[idx], entry_buf, entry_len);
+        return find_at(idx, entry_buf, entry_len);
     }
 
     inline void erase(BYTE* val) {
         int len = bubo_utils::get_entry_len(val);
-        uint32_t idx = hash(val, len) % table_size_;
-
-        Entry** erase_entry = NULL;
-        bool is_spine_entry = false;
-        bool found = find_at(&table_[idx], val, len, &erase_entry, &is_spine_entry);
+        uint32_t index = hash(val, len) % table_size_;
 
+        bool found = find_at(index, val, len);
         if (found) {
-            assert(erase_entry);
-
-            if (is_spine_entry) {
-                // Indicates that value to be removed is in the spine.
-                // The spine may or may not have a chain, which affects the counters.
-                bool deleted = erase_spine_entry(&table_[idx]);
-                if (!deleted) {
-                    // spine did not have a chain. hence decrement table use.
-                    table_curr_use_ --;
-                }
-
-            } else {
-                // erase_entry holds the pointer to the next_ pointer which points to the
-                // Entry to be deleted.
-                erase_next(erase_entry);
-            }
-            num_entries_ --;
+            Entry* entry_to_erase = &table_[index];
```

Comment:
This isn't going to work if the entry to erase is in a slot after `index` due to a collision since you'll erase the entry at `index`, not the one you want.